nextest-runner/src/config/core/imp.rs

```diff
@@ -1124,6 +1150,71 @@ pub(in crate::config) struct NextestConfigImpl {
 }
 
 impl NextestConfigImpl {
+    /// Resolves a profile with inheritance chain
+    fn resolve_profile_chain(&self, profile_name: &str) -> Result<Vec<&CustomProfileImpl>, ProfileNotFound> {
+        let mut visited = HashSet::new();
+        let mut chain = Vec::new();
+        
+        self.resolve_profile_chain_recursive(profile_name, &mut visited, &mut chain)?;
+        Ok(chain)
+    }
+    
+    fn resolve_profile_chain_recursive<'cfg>(
+        &'cfg self,
+        profile_name: &str,
+        visited: &mut HashSet<String>,
+        chain: &mut Vec<&'cfg CustomProfileImpl>,
+    ) -> Result<(), ProfileNotFound> {
+        if visited.contains(profile_name) {
+            return Err(ProfileNotFound::new(
+                profile_name,
+                self.all_profiles().collect::<Vec<_>>(),
+            ));
+        }
+        
+        visited.insert(profile_name.to_string());
+        
+        let profile = self.get_profile(profile_name)?;
+        if let Some(profile) = profile {
+            if let Some(parent_name) = &profile.inherit {
+                self.resolve_profile_chain_recursive(parent_name, visited, chain)?;
+            }
+            chain.push(profile);
+        }
+    
+        Ok(())
+    }
+
+    fn check_inheritance_cycles(&self) -> Result<(), ConfigParseError> {
+        let mut graph = Graph::<String, (), Directed>::new();
+        let mut node_map = HashMap::new();
+
+        for profile in self.all_profiles() {
+            let node = graph.add_node(profile.to_string());
+            node_map.insert(profile.to_string(), node);
+        }
+
+        for (profile_name, profile) in &self.other_profiles {
+            if let Some(inherit_name) = &profile.inherit {
+                if let (Some(&from), Some(&to)) = (node_map.get(inherit_name), node_map.get(profile_name)) {
+                    graph.add_edge(from, to, ());
+                }
+            }
+        }
+
+        match toposort(&graph, None) {
+            Ok(_) => Ok(()),
+            Err(cycle) => {
+                let cycle_profile = graph[cycle.node_id()].clone();
+                Err(ConfigParseError::new(
+                    "Inheritance cycle detected in profile configuration",
```

Comment:
We use lowercase for errors:

```suggestion
                    "inheritance cycle detected in profile configuration",
```